packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -853,6 +882,24 @@ export class Project extends ProjectBase {
     }
   }
 
+  public enableBatchBuilds(): IEnableBatchBuildsReturn | undefined {
+    if (!this._batchServiceRole) {
+      this._batchServiceRole = new iam.Role(this, 'BatchServiceRole', {
+        roleName: PhysicalName.GENERATE_IF_NEEDED,
+        assumedBy: new iam.ServicePrincipal('codebuild.amazonaws.com'),
+      });
+      this._batchServiceRole.addToPrincipalPolicy(new iam.PolicyStatement({
+        resources: [Lazy.string({
```

Comment:
Why use a Lazy here? Why not just `resources: [this.projectArn],`?

Reply:
Because when bind() is called on the sources we haven't calculated the name yet meaning it was undefined. It was a bug I didn't spot earlier.